test/clojure/core_test/conj_bang.cljc

```diff
@@ -0,0 +1,77 @@
+(ns clojure.core-test.conj-bang
+  (:require [clojure.test :refer [are deftest is testing]]
+            [clojure.core-test.portability #?(:cljs :refer-macros :default :refer) [when-var-exists]]))
+
+(when-var-exists conj!
+  (deftest test-conj!
+
+    (testing "conj!"
+      (is (= [] (persistent! (conj!)))))
+
+    (testing "conj! coll"
+      (testing "when coll is transient"
```

Comment:
Can we add a test for the count of the transient vector before coercing it to a persistent vector?

Reply:
Why?

Reply:
Shouldn't we ensure that the transient has the correct size/shape during intermediate calls to `conj!` (or any other `x!` applied to a transient)?